lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -614,6 +614,197 @@ def postprocess_element(self, elem: Element) -> Element:
+        if (element1 is not None) and (element1.type in small) and (element2 is not None) and (element2.type in small):
+            return True
+
+        # Add header to next element (images, tables)
+        if element2.type not in ["Section-header", "Title"]:
+            element2.data["_header"] = element1.data["_header"]
+            if element2.text_representation:
+                if element2.data["_header"]:
+                    element2.text_representation = element2.data["_header"] + "\n" + element2.text_representation
+            else:
+                element2.text_representation = element2.data["_header"]
+        return False
+
+    def merge(self, elt1: Element, elt2: Element) -> Element:
+        """Merge two elements; the new element's fields will be set as:
+            - type: "Section-header", "MergedText"
+            - binary_representation: elt1.binary_representation + elt2.binary_representation
+            - text_representation: elt1.text_representation + elt2.text_representation
+            - bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            - properties: elt1's properties + any of elt2's properties that are not in elt1
+            note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Element): the first element (numbers of tokens in it is stored by `preprocess_element` as element1["token_count"])
+            element2 (Element): the second element (numbers of tokens in it is stored by `preprocess_element` as element2["token_count"])
+
+        Returns:
+            Element: a new merged element from the inputs (and number of tokens in it)
+        """
+
+        tok1 = elt1.data["token_count"]
+        tok2 = elt2.data["token_count"]
+        new_elt = Element()
+
+        if elt1.type in ["Section-header", "Title"] and elt2.type in ["Section-header", "Title"]:
+            new_elt.type = "Section-header"
+        else:
+            new_elt.type = "Text"
+
+        # Merge binary representations by concatenation
+        if elt1.binary_representation is None or elt2.binary_representation is None:
+            new_elt.binary_representation = elt1.binary_representation or elt2.binary_representation
+        else:
+            new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
+
+        # Merge text representations by concatenation with a newline
+        new_elt_text_representation = "\n".join(filter(None, [elt1.text_representation, elt2.text_representation]))
+        new_elt.text_representation = new_elt_text_representation if new_elt_text_representation else None
+        if elt1.text_representation is None or elt2.text_representation is None:
+            new_elt.data["token_count"] = tok1 + tok2
+        else:
+            new_elt.data["token_count"] = tok1 + 1 + tok2
```

Comment:
The logic for calculating the new `token_count` is wrong for elements the merged in an image that started with a summary. It looks like HeaderAugmenterMerger typically throws away the summary and keeps the text_representation which could have a greater or lesser number of tokens than the summary except when we don't merge an Image and leave it by itself. The splitter takes care of this issue so this can be a "won't fix" if you want.

One fix would be to change `preprocess_element`.

Reply:
Rather than having every merger do this, I think it would be better to make the decision of adding summary to text representation in the summarize and summarize_images transform? Right now, the greedy section merger adds summary only of images to text_representation.

Reply:
Update: Based on your slack conversation with @bsowell, it seems like he does not want mergers to use the summary. In that case, just change `preprocess_element`.

Reply:
Are we okay with the fact we currently throw away summaries when we merge images with other elements?

Reply:
> Rather than having every merger do this, I think it would be better to make the decision of adding summary to text representation in the summarize and summarize_images transform? Right now, the greedy section merger adds summary only of images to text_representation.

Ok.